ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v1/container/clientlibs/editor/js/editDialog.js

```diff
@@ -39,9 +39,214 @@
+                addFormParameter(afAssetPath + '/formmodel', schemaType);
+                addFormParameter(afAssetPath + '/schemaRef');
+            }
+            if (schemaType != NONE) {
+                document.querySelector(FORM_MODEL_SELECTOR).disabled = true;
+                dialog.find(SCHEMA_TYPE)[0].removeAttribute("disabled");
+            }
+            document.body.appendChild(formModelChangeConfirmationDialog);
+            prefillSchema(schemaType, dialog);
+        }
+    };
+
+    function selectFormModelOnChanged(dialog) {
+        var schemaTypeSelected = dialog.find(FORM_MODEL_SELECTOR);
+        if(schemaTypeSelected.length > 0){
+            schemaTypeSelected = schemaTypeSelected[0].value;
+            setElementValue(dialog, DAM_SCHEMA_TYPE, schemaTypeSelected)
+            hideContainersExcept(schemaTypeSelected);
+        }
+    };
+
+    function prefillSchema(schemaType, dialog){
+        var schemaRef = dialog.find(SCHEMA_REF);
+        if(schemaRef.length > 0){
+            schemaRef = schemaRef[0].value;
+            setElementValue(dialog, DAM_SCHEMA_REF, schemaRef);
+            if (schemaType == JSON_SCHEMA) {
+                $(SCHEMA_DROPDOWN_SELECTOR).val(schemaRef);
+            } else if (schemaType == FORM_DATA_MODEL) {
+                $(FDM_DROPDOWN_SELECTOR).val(schemaRef);
+            }
+        }
+    };
+
+    function schemaSelectorOnChanged(dialog) {
+        var selectedSchema = dialog.find(SCHEMA_DROPDOWN_SELECTOR);
+        if(selectedSchema.length > 0){
+            selectedSchema = selectedSchema[0].value;
+            setElementValue(dialog, SCHEMA_REF, selectedSchema);
+            setElementValue(dialog, DAM_SCHEMA_REF, selectedSchema);
+            if (configuredFormModel) {
+                confirmFormModelChange(selectedSchema, $(SCHEMA_DROPDOWN_SELECTOR));
+            } else {
+                toBeConfiguredFormModel = selectedSchema;
+            }
+        }
+    };
+
+    function fdmSelectorOnChanged(dialog) {
+        var selectedSchema = dialog.find(FDM_DROPDOWN_SELECTOR);
+        if(selectedSchema.length > 0) {
+            selectedSchema = selectedSchema[0].value;
+            setElementValue(dialog, SCHEMA_REF, selectedSchema);
+            setElementValue(dialog, DAM_SCHEMA_REF, selectedSchema);
+            if (configuredFormModel) {
+                confirmFormModelChange(selectedSchema, $(FDM_DROPDOWN_SELECTOR));
+            } else {
+                toBeConfiguredFormModel = selectedSchema;
+            }
+        }
+    };
+
+    function setElementValue(dialog, elementRef, value){
+        var element = dialog.find(elementRef);
+        if(element.length > 0){
+            element[0].value = value;
+        }
+    }
+
+    function getAfAssetMetadataPath() {
+        return window.guidelib.author.AuthorUtils.getGuideContainerPath().replace(FM_AF_ROOT, FM_DAM_ROOT).replace("/guideContainer", "/metadata");
```

Comment:
are we loading AuthorUtils clientlibs in core components?

Reply:
it is present there